e4tools/bundles/org.eclipse.e4.tools.emf.ui/src/org/eclipse/e4/tools/emf/ui/internal/common/component/ControlFactory.java

```diff
@@ -745,12 +745,14 @@ public static <M> void createCheckBox(Composite parent, String label, IObservabl
 	public static <M> void createCheckBox(Composite parent, String label, String tooltip, IObservableValue<M> master,
 			EMFDataBindingContext context, IWidgetValueProperty<Button, Boolean> selectionProp,
 			IValueProperty<? super M, Boolean> modelProp) {
+		final Label l = new Label(parent, SWT.NONE);
```

Comment:
```suggestion
		final Label label = new Label(parent, SWT.NONE);
```

Reply:
It is not possible to make the reviewed change without causing the code to have an error because naming the `Label` `label` would shadow the parameter `label` of type `String` passed into the function.

Since naming the variable of type `Label` `l` was also done by other people who wrote similar functions, for example in the functions `createXMIId` and `createTextField` in the same file `ControlFactory.java`, I was wondering if a change needs to be made.

Reply:
That's right. This is an unfortunate situation and done like this in similar cases in this class.
So I think it's fine.